### PR TITLE
Sampler ycbcr conversion

### PR DESCRIPTION
**Pull request by Rua**
```markdown
- Added `SamplerYcbcrConversion`, and the ability to attach it to a sampler or image view.
```

Depends on #1806.

A sampler YCbCr conversion is a Vulkan object that's used for sampling video frames. It's a bit more of a niche feature, and I myself have no use case for it, but it's just one more thing that brings Vulkano closer to being complete. It's also needed for video encoding/decoding, a still-experimental Vulkan feature that was requested in #1756. I don't know if it's wise to implement an experimental feature, but if we do, then the requirements are now done.

### CODE REVIEW

**Comment by Eliah-Lakhin** [comment]
@Rua After the #1806 merge, there are merging conflicts.

File: vulkano/src/sampler/ycbcr.rs [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+//! A sampler YCbCr conversion is an object that assists a sampler when converting from YCbCr
+//! formats and/or YCbCr texel input data. It is used to read frames of video data within a shader,
+//! possibly to apply it as texture on a rendered primitive. Sampler YCbCr conversion can only be
+//! used with certain formats, and conversely, some formats require the use of a sampler YCbCr
+//! conversion to be sampled at all.
+//!
+//! A sampler YCbCr conversion can only be used with a combined image sampler descriptor in a
+//! descriptor set. The conversion must be attached on both the image view and sampler in the
+//! descriptor, and the sampler must be included in the descriptor set layout as an immutable
+//! sampler.
```
Comment (Eliah-Lakhin) [inline vulkano/src/sampler/ycbcr.rs:21]: Maybe provide an example of the sampler use?

Comment (Rua) [reply vulkano/src/sampler/ycbcr.rs:21]: Ok, how is this?

Comment (Eliah-Lakhin) [reply vulkano/src/sampler/ycbcr.rs:21]: Excellent!

**Comment by Eliah-Lakhin** [comment]
@Rua Merged :+1:

**Pull request merged**